lib/global-error/src/macros.rs

```diff
@@ -121,17 +212,90 @@ macro_rules! unwrap_ref {
 }
 pub use unwrap_ref;
 
+/// `AssertionErrorUnwrap` is a trait used for handling unwrapping of `Result`
+/// and `Option` types.
+///
+/// This trait provides a method `assertion_error_unwrap` that takes a `Result`
+/// or `Option` and returns a `Result`. If the original `Result` or `Option` is
+/// `Ok` or `Some`, it returns `Ok` with the unwrapped value. If it's `Err` or
+/// `None`, it returns `Err` with an `AssertionError::Unwrap` that includes a
+/// message and location.
+pub trait AssertionErrorUnwrap {
+	type WrappedType;
+
+	fn assertion_error_unwrap(
+		self,
+		message: &'static str,
+	) -> Result<Self::WrappedType, crate::ext::AssertionError>;
+}
+
+impl<T, E: core::fmt::Debug> AssertionErrorUnwrap for Result<T, E> {
+	type WrappedType = T;
+
+	/// Unwraps a `Result` or `Option`.
+	///
+	/// # Arguments
+	///
+	/// * `self` - The `Result` or `Option` to unwrap.
+	/// * `message` - A message to include in the `AssertionError::Unwrap` if
+	///   `self` is `Err` or `None`.
+	///
+	/// # Returns
+	///
+	/// * `Ok` with the unwrapped value if `self` is `Ok` or `Some`.
+	/// * `Err` with an `AssertionError::Unwrap` that includes `message` and the
+	///   location if `self` is `Err` or `None`.
+	fn assertion_error_unwrap(
+		self,
+		message: &'static str,
+	) -> Result<T, crate::ext::AssertionError> {
+		match self {
+			Ok(t) => Ok(t),
+			Err(e) => Err(crate::ext::AssertionError::Unwrap {
+				message, // format!("{}: {:?}", message, e).as_str(),
+				location: crate::location!(),
```

Comment:
This will always have the location `lib/global-error/src/macros.rs:256:xxx`. To have it be the location of where the `unwrap` macro/function was called, it has to be passed in as an argument with `message`

Example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=08fd9e7b7f98235c94c4af72e6210dca

Reply:
Awesome catch, I hadn't considered this. Thanks!